tests/unit_tests/test_scoring_hpf_scoring.py

```diff
@@ -1,35 +1,152 @@
 """Test functionalities of hpf scoring."""
 
 from src.scoring.hpf_scoring import HPFScoring
+from src.data_store.local_data_store import LocalDataStore
 from edward.models import Poisson
 import tensorflow as tf
+import numpy as np
+import unittest
 
 
-def test_basic_object():
-    """Test basic HPF object without datastore."""
-    hpf = HPFScoring()
-    assert hpf is not None
-    assert hpf.theta is None
-    assert round(float(hpf.epsilon), 9) == 0.000156564
-    assert hpf.theta_dummy is not None
-    assert isinstance(hpf.theta_dummy, Poisson)
-    assert hpf.theta_dummy.shape.dims[0].value == 13
-
-
-def test_model_details():
-    """Test the basic model details function."""
-    hpf = HPFScoring()
-    assert hpf is not None
-    details = """The model will be scored against
-        0 Packages,
-        0 Manifests,
-        Theta matrix of size 1.52587890625e-05 MB, and
-        Beta matrix of size 1.52587890625e-05 MB."""
-    assert hpf.model_details() == details
-
-
-def test_get_sizeof():
-    """Test static _getsizeof method."""
-    int_value = 1
-    int_size = 2.6702880859375e-05
-    assert HPFScoring._getsizeof(int_value) == "{} MB".format(int_size)
+class TestHPFScoringMethods(unittest.TestCase):
+    """Test functionalities of hpf scoring."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialise the local data store and HPF object."""
+        super(TestHPFScoringMethods, self).__init__(*args, **kwargs)
+        self.local_obj = LocalDataStore("tests/test_data")
+        self.hpf_obj = HPFScoring(self.local_obj)
+
+    def test_basic_object(self):
+        """Test basic HPF object."""
+        assert self.hpf_obj is not None
+        assert self.hpf_obj.theta is not None
+        assert self.hpf_obj.beta is not None
+        assert round(float(self.hpf_obj.epsilon), 9) == 0.000156564
+        assert self.hpf_obj.theta_dummy is not None
+        assert isinstance(self.hpf_obj.theta_dummy, Poisson)
+        assert self.hpf_obj.theta_dummy.shape.dims[0].value == 13
+
+    def test_normalise_result(self):
+        """Test normlise_result() with dummy result."""
+        result = np.array([[0.1, 0.2, 0.3],
+                           [0.1, 0.2, 0.3],
+                           [0.1, 0.2, 0.3]])
+        input_id_set = {1}
+        normal = self.hpf_obj.normalize_result(
+            result, input_id_set, len(result))
+        self.assertListEqual(normal.tolist(),
+                             [0.20000000000000004,
+                              -1.0,
+                              0.20000000000000004])
```

Comment:
is the computation that stable? I'd suggest to use some form of epsilon check

Reply:
Yes, we can do that, I can also perform some rounding off.

Reply:
thanks a lot in advance @sara-02